app/src/main/java/com/example/playlistmakervmp/SearchActivity.kt

```diff
@@ -177,6 +228,66 @@ class SearchActivity : AppCompatActivity() {
             placeholderNothingFoundText.visibility = View.GONE
             searchTracks()
         }
+
+        trackAdapter.setOnTrackClickListener(object : OnTrackClickListener {
+            override fun onTrackClick(position: Int) {
+                val editor = sharedPrefs.edit()
+                if ((history.size <= 9) and !isTrackInHistory(tracks[position])) {
+                    history.add(0, tracks[position])
+                    json = Gson().toJsonTree(history).asJsonArray.toString()
+                    editor.putString("TRACKS", json).apply()
+                }
+                else if ((history.size == 10) and !isTrackInHistory(tracks[position])) {
+
+                    history.add(0, tracks[position])
+                    history.removeLast()
+                    json = Gson().toJsonTree(history).asJsonArray.toString()
+                    editor.putString("TRACKS", json).apply()
+                }
+                else if (isTrackInHistory(tracks[position])) {
+
+                    lateinit var songbuf: Track
+                    for (song in history) {
+                        if (song.trackId == tracks[position].trackId) {
+                            songbuf = song
+                        }
+                    }
+                    history.remove(songbuf)
+                    history.add(0, tracks[position])
+                    json = Gson().toJsonTree(history).asJsonArray.toString()
+                    editor.putString("TRACKS", json).apply()
+                }
+            }
+        })
+        editText.setOnFocusChangeListener { view, hasFocus ->
+            if (history.size > 0) {
```

Comment:
```suggestion
            if (history.isNotEmpty()) {
```
Так будет немного читабельнее